jmx/src/main/java/org/jboss/as/jmx/model/ModelControllerMBeanServerPlugin.java

```diff
@@ -246,20 +266,47 @@ public void removeNotificationListener(ObjectName name, NotificationListener lis
         notificationRegistry.unregisterNotificationHandler(pathAddress, handler, handler);
     }
 
+    @Override
     public void removeNotificationListener(ObjectName name, NotificationListener listener) throws InstanceNotFoundException,
             ListenerNotFoundException {
         removeNotificationListener(name, listener, null, null);
     }
 
+    @Override
     public void removeNotificationListener(ObjectName name, ObjectName listener, javax.management.NotificationFilter filter, Object handback)
             throws InstanceNotFoundException, ListenerNotFoundException {
         throw new RuntimeOperationsException(JmxLogger.ROOT_LOGGER.removeNotificationListerWithObjectNameNotSupported(listener));
     }
 
+    @Override
     public void removeNotificationListener(ObjectName name, ObjectName listener) throws InstanceNotFoundException, ListenerNotFoundException {
         throw new RuntimeOperationsException(JmxLogger.ROOT_LOGGER.removeNotificationListerWithObjectNameNotSupported(listener));
     }
 
+    @Override
+    public JmxTarget getJmxTarget(final ObjectName name, final String method) {
+        ModelControllerMBeanHelper helper = getHelper(name);
+        PathAddress address = name != null ? helper.resolvePathAddress(name) : PathAddress.EMPTY_ADDRESS;
+        return new JmxTarget(method, name, address, helper.resolveConstraints(address));
+    }
+
+    @Override
+    public JmxAction getJmxAction(final ObjectName name, final String method, String operation, JmxAction.Impact impact, boolean isNonFacadeMBeansSensitive, String attributeName) {
```

Comment:
This should just throw UnsupportedOperationException. This class returns false from shouldAuthorize(), so it shouldn't attempt to do authorization tasks.

For the JMX domains handled by this MBeanServerPlugin, the standard authz from the ModelController, OperationContext etc will happen, so there is no anything in the JMX layer.

With that change, a whole lot of stuff can presumably come out.